public_html/wikiWhoProxy.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+header( 'Content-Type: application/json' );
+header( 'Access-Control-Allow-Origin: *' );
+
+// Restrict access to Wikipedias.
+if ( 1 !== preg_match( '/^https:\/\/.*?\.wikipedia\.org/', $_SERVER[ 'HTTP_REFERER' ] ) ) {
+    http_response_code( 403 );
+    die();
+}
+
+$cnf = parse_ini_file( '../wikiwho.ini' );
+if ( !$cnf || !$cnf[ 'user' ] || !$cnf[ 'password' ] ) {
+    throw new Exception(
+        'WikiWho API credentials not found at ' . dirname( __FILE__, 2 ) . '/wikiwho.ini'
```

Comment:
The two different ways of constructing the path to wikiwho.ini confused me. Maybe a var?

Reply:
Done!